packages/lesswrong/server/migrations/2020-06-08-clearOldPartiallyReadSequences.ts

```diff
@@ -0,0 +1,48 @@
+import { registerMigration, forEachDocumentBatchInCollection } from './migrationUtils';
+import { Users } from '../../lib/collections/users/collection';
+import * as _ from 'underscore';
+
+const excludeOld = (partiallyReadSequences: Array<any>, dateCutoff: Date): Array<any> => {
+  return _.filter(partiallyReadSequences, s=>s.lastReadTime >= dateCutoff);
+}
+
+registerMigration({
+  name: "clearOldPartiallyReadSequences",
+  dateWritten: "2020-06-08",
+  idempotent: true,
+  action: async () => {
+    const thirtyDaysInMs = 30*24*60*60*1000;
```

Comment:
Doublechecking what "30 days" means here exactly. If, in the past week, I accidentally clicked on a some sequence post and it got added to the partiallyReadSequences list, will that still show up here? 

I'm guessing so. I know I've personally clicked on 4 sequence posts in the past 30 days which are now in my continue reading, all of which were unintentional. If it were possible to filter for people who have actively clicked on continue reading in the past 30 days, that'd seem better to me. (I think we have this data around somewhere but it might be trickier to use. Ask Ruby?)

But I think it's fairly important for most users to get an actual clean slate of continue reading.

Reply:
My current model is indeed that pushing the new frontpage rework would still be blocked on doing this more in-depth migration, but that this shouldn't block this PR from getting merged, since it's still a net improvement. 

My guess is we should just ask Ruby for a list of people who have clicked on a continue reading button in the last 30 days, and additionally reset the continue reading section for anyone who hasn't done so.

Reply:
I agree the PR doesn't need the in-depth migration, but if the current migration does what I think it does, I think it probably isn't the right migration, and doesn't make sense to include that bit. Maybe doesn't need to be a blocking comment (I'm a bit confused about what things should be blocking or not)

Reply:
I prefer to merge the current migration, since if we merge this with the migration, in case we get distracted by some other thing, we can just run this migration in 30 days and get the same effect without having to do anything complicated.

Reply:
Ah, fair.